modules/apps/server/server-admin-web-test/src/testIntegration/java/com/liferay/server/admin/web/internal/portlet/action/test/EditServerMVCActionCommandTest.java

```diff
@@ -209,6 +232,133 @@ private LayoutRevision _getLayoutRevision() throws Exception {
+			mockLiferayPortletActionRequest.addParameter(
+				entry.getKey(), entry.getValue());
+		}
+
+		ReflectionTestUtil.invoke(
+			_mvcActionCommand, "_updateMail",
+			new Class<?>[] {
+				ActionRequest.class, javax.portlet.PortletPreferences.class
+			},
+			mockLiferayPortletActionRequest, portletPreferences);
+
+		Assert.assertEquals(
+			parameters.get("advancedProperties"),
+			portletPreferences.getValue(
+				PropsKeys.MAIL_SESSION_MAIL_ADVANCED_PROPERTIES, null));
+		Assert.assertEquals(
+			parameters.get("pop3Host"),
+			portletPreferences.getValue(
+				PropsKeys.MAIL_SESSION_MAIL_POP3_HOST, null));
+		Assert.assertEquals(
+			parameters.get("pop3Password"),
+			portletPreferences.getValue(
+				PropsKeys.MAIL_SESSION_MAIL_POP3_PASSWORD, null));
+		Assert.assertEquals(
+			parameters.get("pop3Port"),
+			portletPreferences.getValue(
+				PropsKeys.MAIL_SESSION_MAIL_POP3_PORT, null));
+		Assert.assertEquals(
+			Boolean.valueOf(parameters.get("pop3Secure")) ?
+				Account.PROTOCOL_POPS : Account.PROTOCOL_POP,
+			portletPreferences.getValue(
+				PropsKeys.MAIL_SESSION_MAIL_STORE_PROTOCOL, null));
+		Assert.assertEquals(
+			parameters.get("pop3User"),
+			portletPreferences.getValue(
+				PropsKeys.MAIL_SESSION_MAIL_POP3_USER, null));
+		Assert.assertEquals(
+			parameters.get("popServerNotificationsEnabled"),
+			portletPreferences.getValue(
+				PropsKeys.POP_SERVER_NOTIFICATIONS_ENABLED, null));
+		Assert.assertEquals(
+			parameters.get("smtpHost"),
+			portletPreferences.getValue(
+				PropsKeys.MAIL_SESSION_MAIL_SMTP_HOST, null));
+		Assert.assertEquals(
+			parameters.get("smtpPassword"),
+			portletPreferences.getValue(
+				PropsKeys.MAIL_SESSION_MAIL_SMTP_PASSWORD, null));
+		Assert.assertEquals(
+			parameters.get("smtpPort"),
+			portletPreferences.getValue(
+				PropsKeys.MAIL_SESSION_MAIL_SMTP_PORT, null));
+		Assert.assertEquals(
+			Boolean.valueOf(parameters.get("smtpSecure")) ?
+				Account.PROTOCOL_SMTPS : Account.PROTOCOL_SMTP,
+			portletPreferences.getValue(
+				PropsKeys.MAIL_SESSION_MAIL_TRANSPORT_PROTOCOL, null));
+		Assert.assertEquals(
+			parameters.get("smtpStartTLSEnable"),
+			portletPreferences.getValue(
+				PropsKeys.MAIL_SESSION_MAIL_SMTP_STARTTLS_ENABLE, null));
+		Assert.assertEquals(
+			parameters.get("smtpUser"),
+			portletPreferences.getValue(
+				PropsKeys.MAIL_SESSION_MAIL_SMTP_USER, null));
+	}
+
+	private void _testUpdateMailPortletPreferences(long companyId)
+		throws Exception {
+
+		javax.portlet.PortletPreferences portletPreferences =
+			PrefsPropsUtil.getPreferences(companyId);
+
+		try {
+			_testUpdateMailPortletPreferences(
+				RandomTestUtil::randomBoolean, RandomTestUtil::randomInt,
+				RandomTestUtil::randomString, portletPreferences);
+
+			_testUpdateMailPortletPreferences(
+				() -> Boolean.FALSE, () -> 0, () -> StringPool.BLANK,
+				portletPreferences);
+		}
+		finally {
+			for (String key : new String[0]) {
+				portletPreferences.reset(key);
+			}
+		}
```

Comment:
@marianoalvarosaiz, the code `portletPreferences.reset(key);` can't be executed for now.  For the reset part, we may use the below way.

For system setting, it doesn't exist data in the database by default. So we may use PortalPreferencesLocalService to delete the generated portalpreferences data. For example,

```
PortalPreferencesWrapper portalPreferencesWrapper = (PortalPreferencesWrapper)portletPreferences;

PortalPreferencesImpl portalPreferencesImpl = portalPreferencesWrapper.getPortalPreferencesImpl();

PortalPreferences portalPreferences = PortalPreferencesLocalService.fetchPortalPreferences(portalPreferencesImpl.getOwnerId(), portalPreferencesImpl.getOwnerType());

PortalPreferencesLocalService.delete(portalPreferences);
```
For the above code fragment, you may place it to finally blcok in the _testUpdateMailPortletPreferences() 

For the instance setting, we may use new company and then delete the company in the testUpdateMailCompanyPortletPreferences().